### DISCUSSION TITLE
Introducing a shorthand directive x-itext and x-ihtml

### DISCUSSION THREAD

**Discussion by simonferndriger** [rehhouari/alpinejs-i18n#3] [Ideas]
This addition would allow to ommit the always present `$t('...')` inside the x-text/html attributes and would make it easier to read:

`x-itext="welcome"` and `x-itext="('welcome', { name: 'John' })"` for variables.

Also, why don't you make it an official Alpine plugin?

**Comment by rehhouari** [comment]
hey, thanks for the suggestion!

1. `x-text` & `x-html` are core directives, to make their I18n equivilant (e.g., `x-itext`), I'd have to copy code from Alpine.js source, which is alot of extra code just for a shorthand.
2. I can't make it an official plugin because I'm not the Alpine.js developer, That is @/calebporzio. Also, I highly doubt he'd be interested in adding it to the core, because i18n is considered a secondary concern for most people.

This is why I cant add that shorthand, atleast for now.

**Reply by simonferndriger** [reply]
Thanks, makes sense for the current situation. However, who knows what the future might bring 😉
